### PR TITLE
bbb-config: valdiate hostname using regex

### PR DESCRIPTION
**Pull request by Stadicus**
According to IETF RFC 952 (https://tools.ietf.org/html/rfc952),

    A "name" (Net, Host, Gateway, or Domain name) is a text string up to
    24 characters drawn from the alphabet (A-Z), digits (0-9), minus sign (-),
    and period (.). Note that periods are only allowed when they serve to
    delimit components of "domain style names". No blank or space characters
    are permitted as part of a name. No distinction is made between upper and
    lower case. The first character must be an alpha character. The last
    character must not be a minus sign or period.

Using the following regex to validate hostname before change:

    ^[a-z][a-z0-9-]{0,22}[a-z0-9]$

### CODE REVIEW

**Comment by 0xB10C** [comment]
To slow.

**Comment by Stadicus** [comment]
From what I understand, a period is only allowed _"when they serve to delimit components of "domain style names"._ 

Do you think that would be a legit hostname? I don't care either way.

**Comment by 0xB10C** [comment]
_domain style name_ probably includes hostnames. (_`A "name" (Net, Host, Gateway, or Domain name)`_)

Yet I'm not sure how often people give _domain style names_ to their home devices e.g. `bitcoin.box`...

It makes the regex more complicated and since we are probably deriving the hostname from the name the user gives his BBB this might not need to be changed...

I'd imagine the hostname from name derivation being something along the lines of `toLowercase()` and `replace(" ", "-")`. I don't see how periods would appear there.
```
# name         ->       hostname
Satoshis Base  ->       satoshis-base
Bitcoin Box    ->       bitcoin-box
```

tl;dr: I guess it's fine as is now.
